File: model/src/main/java/ru/sberleasing/model/persistance/DatabaseModule.java
package ru.sberleasing.model.persistance;

import android.arch.persistence.room.Room;
import android.content.Context;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class DatabaseModule {

    @Provides
    @Singleton
    Database provideDatabase(Context context) {
        return Room.databaseBuilder(context, Database.class, "maindb")
                .fallbackToDestructiveMigration()
                .build();
    }

    @Provides
    @Singleton
    UserDao provideUserDao(Database database) {
        return database.userDao();
    }

}

File: model/src/main/java/com/example/model/persistance/DatabaseModule.kt
package com.example.model.persistance

import android.arch.persistence.db.SupportSQLiteDatabase
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context
import dagger.Module
import dagger.Provides
import java.util.concurrent.Executors

import javax.inject.Singleton

@Module
class DatabaseModule {

    @Provides
    @Singleton
    internal fun provideDatabase(context: Context): Database {
        return Room.databaseBuilder(context, Database::class.java, "maindb")
                .fallbackToDestructiveMigration()
                .addCallback(object: RoomDatabase.Callback() {
                    override fun onCreate(db: SupportSQLiteDatabase) {
                        Executors.newSingleThreadExecutor().execute({
                            provideDatabase(context).cityDao().apply {
                                insertAll(seedCities()) // TODO seeding done wrong
                            }
                        })
                    }
                })
                .build()
    }

    @Provides
    @Singleton
    internal fun provideCityDao(database: Database): CityDao {
        return database.cityDao()
    }

}
